<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Christopher Eeles" />
  <meta name="dcterms.date" content="2021-08-05" />
  <title>PCOSP: Pancreatic Cancer Overall Survival Predictor</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="PancreaticCancerOverallSurvivalPredictor_files/reveal.js-3.3.0.1/css/reveal.css"/>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="PancreaticCancerOverallSurvivalPredictor_files/reveal.js-3.3.0.1/css/theme/simple.css" id="theme">


  <!-- some tweaks to reveal css -->
  <style type="text/css">
    .reveal h1 { font-size: 2.0em; }
    .reveal h2 { font-size: 1.5em;  }
    .reveal h3 { font-size: 1.25em;	}
    .reveal h4 { font-size: 1em;	}

    .reveal .slides>section,
    .reveal .slides>section>section {
      padding: 0px 0px;
    }



    .reveal table {
      border-width: 1px;
      border-spacing: 2px;
      border-style: dotted;
      border-color: gray;
      border-collapse: collapse;
      font-size: 0.7em;
    }

    .reveal table th {
      border-width: 1px;
      padding-left: 10px;
      padding-right: 25px;
      font-weight: bold;
      border-style: dotted;
      border-color: gray;
    }

    .reveal table td {
      border-width: 1px;
      padding-left: 10px;
      padding-right: 25px;
      border-style: dotted;
      border-color: gray;
    }


  </style>

    <style type="text/css">code{white-space: pre;}</style>


<!-- Printing and PDF exports -->
<script id="paper-css" type="application/dynamic-css">

/* Default Print Stylesheet Template
   by Rob Glazebrook of CSSnewbie.com
   Last Updated: June 4, 2008
   Feel free (nay, compelled) to edit, append, and
   manipulate this file as you see fit. */


@media print {

	/* SECTION 1: Set default width, margin, float, and
	   background. This prevents elements from extending
	   beyond the edge of the printed page, and prevents
	   unnecessary background images from printing */
	html {
		background: #fff;
		width: auto;
		height: auto;
		overflow: visible;
	}
	body {
		background: #fff;
		font-size: 20pt;
		width: auto;
		height: auto;
		border: 0;
		margin: 0 5%;
		padding: 0;
		overflow: visible;
		float: none !important;
	}

	/* SECTION 2: Remove any elements not needed in print.
	   This would include navigation, ads, sidebars, etc. */
	.nestedarrow,
	.controls,
	.fork-reveal,
	.share-reveal,
	.state-background,
	.reveal .progress,
	.reveal .backgrounds {
		display: none !important;
	}

	/* SECTION 3: Set body font face, size, and color.
	   Consider using a serif font for readability. */
	body, p, td, li, div {
		font-size: 20pt!important;
		font-family: Georgia, "Times New Roman", Times, serif !important;
		color: #000;
	}

	/* SECTION 4: Set heading font face, sizes, and color.
	   Differentiate your headings from your body text.
	   Perhaps use a large sans-serif for distinction. */
	h1,h2,h3,h4,h5,h6 {
		color: #000!important;
		height: auto;
		line-height: normal;
		font-family: Georgia, "Times New Roman", Times, serif !important;
		text-shadow: 0 0 0 #000 !important;
		text-align: left;
		letter-spacing: normal;
	}
	/* Need to reduce the size of the fonts for printing */
	h1 { font-size: 28pt !important;  }
	h2 { font-size: 24pt !important; }
	h3 { font-size: 22pt !important; }
	h4 { font-size: 22pt !important; font-variant: small-caps; }
	h5 { font-size: 21pt !important; }
	h6 { font-size: 20pt !important; font-style: italic; }

	/* SECTION 5: Make hyperlinks more usable.
	   Ensure links are underlined, and consider appending
	   the URL to the end of the link for usability. */
	a:link,
	a:visited {
		color: #000 !important;
		font-weight: bold;
		text-decoration: underline;
	}
	/*
	.reveal a:link:after,
	.reveal a:visited:after {
		content: " (" attr(href) ") ";
		color: #222 !important;
		font-size: 90%;
	}
	*/


	/* SECTION 6: more reveal.js specific additions by @skypanther */
	ul, ol, div, p {
		visibility: visible;
		position: static;
		width: auto;
		height: auto;
		display: block;
		overflow: visible;
		margin: 0;
		text-align: left !important;
	}
	.reveal pre,
	.reveal table {
		margin-left: 0;
		margin-right: 0;
	}
	.reveal pre code {
		padding: 20px;
		border: 1px solid #ddd;
	}
	.reveal blockquote {
		margin: 20px 0;
	}
	.reveal .slides {
		position: static !important;
		width: auto !important;
		height: auto !important;

		left: 0 !important;
		top: 0 !important;
		margin-left: 0 !important;
		margin-top: 0 !important;
		padding: 0 !important;
		zoom: 1 !important;

		overflow: visible !important;
		display: block !important;

		text-align: left !important;
		-webkit-perspective: none;
		   -moz-perspective: none;
		    -ms-perspective: none;
		        perspective: none;

		-webkit-perspective-origin: 50% 50%;
		   -moz-perspective-origin: 50% 50%;
		    -ms-perspective-origin: 50% 50%;
		        perspective-origin: 50% 50%;
	}
	.reveal .slides section {
		visibility: visible !important;
		position: static !important;
		width: auto !important;
		height: auto !important;
		display: block !important;
		overflow: visible !important;

		left: 0 !important;
		top: 0 !important;
		margin-left: 0 !important;
		margin-top: 0 !important;
		padding: 60px 20px !important;
		z-index: auto !important;

		opacity: 1 !important;

		page-break-after: always !important;

		-webkit-transform-style: flat !important;
		   -moz-transform-style: flat !important;
		    -ms-transform-style: flat !important;
		        transform-style: flat !important;

		-webkit-transform: none !important;
		   -moz-transform: none !important;
		    -ms-transform: none !important;
		        transform: none !important;

		-webkit-transition: none !important;
		   -moz-transition: none !important;
		    -ms-transition: none !important;
		        transition: none !important;
	}
	.reveal .slides section.stack {
		padding: 0 !important;
	}
	.reveal section:last-of-type {
		page-break-after: avoid !important;
	}
	.reveal section .fragment {
		opacity: 1 !important;
		visibility: visible !important;

		-webkit-transform: none !important;
		   -moz-transform: none !important;
		    -ms-transform: none !important;
		        transform: none !important;
	}
	.reveal section img {
		display: block;
		margin: 15px 0px;
		background: rgba(255,255,255,1);
		border: 1px solid #666;
		box-shadow: none;
	}

	.reveal section small {
		font-size: 0.8em;
	}

}
</script>


<script id="pdf-css" type="application/dynamic-css">

/**
 * This stylesheet is used to print reveal.js
 * presentations to PDF.
 *
 * https://github.com/hakimel/reveal.js#pdf-export
 */

* {
	-webkit-print-color-adjust: exact;
}

body {
	margin: 0 auto !important;
	border: 0;
	padding: 0;
	float: none !important;
	overflow: visible;
}

html {
	width: 100%;
	height: 100%;
	overflow: visible;
}

/* Remove any elements not needed in print. */
.nestedarrow,
.reveal .controls,
.reveal .progress,
.reveal .playback,
.reveal.overview,
.fork-reveal,
.share-reveal,
.state-background {
	display: none !important;
}

h1, h2, h3, h4, h5, h6 {
	text-shadow: 0 0 0 #000 !important;
}

.reveal pre code {
	overflow: hidden !important;
	font-family: Courier, 'Courier New', monospace !important;
}

ul, ol, div, p {
	visibility: visible;
	position: static;
	width: auto;
	height: auto;
	display: block;
	overflow: visible;
	margin: auto;
}
.reveal {
	width: auto !important;
	height: auto !important;
	overflow: hidden !important;
}
.reveal .slides {
	position: static;
	width: 100%;
	height: auto;

	left: auto;
	top: auto;
	margin: 0 !important;
	padding: 0 !important;

	overflow: visible;
	display: block;

	-webkit-perspective: none;
	   -moz-perspective: none;
	    -ms-perspective: none;
	        perspective: none;

	-webkit-perspective-origin: 50% 50%; /* there isn't a none/auto value but 50-50 is the default */
	   -moz-perspective-origin: 50% 50%;
	    -ms-perspective-origin: 50% 50%;
	        perspective-origin: 50% 50%;
}

.reveal .slides section {
	page-break-after: always !important;

	visibility: visible !important;
	position: relative !important;
	display: block !important;
	position: relative !important;

	margin: 0 !important;
	padding: 0 !important;
	box-sizing: border-box !important;
	min-height: 1px;

	opacity: 1 !important;

	-webkit-transform-style: flat !important;
	   -moz-transform-style: flat !important;
	    -ms-transform-style: flat !important;
	        transform-style: flat !important;

	-webkit-transform: none !important;
	   -moz-transform: none !important;
	    -ms-transform: none !important;
	        transform: none !important;
}

.reveal section.stack {
	margin: 0 !important;
	padding: 0 !important;
	page-break-after: avoid !important;
	height: auto !important;
	min-height: auto !important;
}

.reveal img {
	background:none; border:none; box-shadow:none;
}

.reveal .roll {
	overflow: visible;
	line-height: 1em;
}

/* Slide backgrounds are placed inside of their slide when exporting to PDF */
.reveal section .slide-background {
	display: block !important;
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	z-index: -1;
}

/* All elements should be above the slide-background */
.reveal section>* {
	position: relative;
	z-index: 1;
}

/* Display slide speaker notes when 'showNotes' is enabled */
.reveal .speaker-notes-pdf {
	display: block;
	width: 100%;
	max-height: none;
	left: auto;
	top: auto;
	z-index: 100;
}

/* Display slide numbers when 'slideNumber' is enabled */
.reveal .slide-number-pdf {
	display: block;
	position: absolute;
	font-size: 14px;
}

</script>


<script>
var style = document.createElement( 'style' );
style.type = 'text/css';
var style_script_id = window.location.search.match( /print-pdf/gi ) ? 'pdf-css' : 'paper-css';
var style_script = document.getElementById(style_script_id).text;
style.innerHTML = style_script;
document.getElementsByTagName('head')[0].appendChild(style);
</script>

    <script src="PancreaticCancerOverallSurvivalPredictor_files/header-attrs-2.9/header-attrs.js"></script>
    <script src="PancreaticCancerOverallSurvivalPredictor_files/core-js-2.5.3/shim.min.js"></script>
    <script src="PancreaticCancerOverallSurvivalPredictor_files/react-17.0.0/react.min.js"></script>
    <script src="PancreaticCancerOverallSurvivalPredictor_files/react-17.0.0/react-dom.min.js"></script>
    <script src="PancreaticCancerOverallSurvivalPredictor_files/reactwidget-1.0.0/react-tools.js"></script>
    <script src="PancreaticCancerOverallSurvivalPredictor_files/htmlwidgets-1.5.3/htmlwidgets.js"></script>
    <script src="PancreaticCancerOverallSurvivalPredictor_files/reactable-binding-0.2.3/reactable.js"></script>
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section data-background='images/bhklogo.png' data-background-size='20%' data-background-position='bottom left'>
    <h1 class="title"><b>PCOSP: Pancreatic Cancer Overall Survival Predictor</b></h1>
    <h2 class="author">Christopher Eeles</h2>
    <h3 class="date">2021-08-05</h3>
</section>

<section>
<section id="introduction" class="title-slide slide level1" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h1 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Introduction</strong></h1>
<style>
body { background: #E9EDEE }
</style>
<!--
## **Who Am I?** { data-background='images/bhklogo.png' data-background-size='20%' data-background-position='bottom left'}

* Christopher Eeles, a software developer in the Benjmain Haibe-Kains Lab
  * At Princess Margaret Cancer Center
  * Part of the University Health Network
  * In Toronto, ON, Canada
  * Check out our work at https://bhklab.ca/


## **What Do I Do?** { data-background='images/bhklogo.png' data-background-size='20%' data-background-position='bottom left'}
* Build software tools (mostly R and Python packages) to help researchers and clinicians gain insights from omics data
* Write automated and reproducible bioinformatics analysis pipelines
* Design scalable ETL workflows to process raw published omics data into curated data sets and databases
-->
</section>
<section id="what-are-we-discussing-today" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>What Are We Discussing Today?</strong></h2>
<ul>
<li class="fragment">Pancreatic Cancer Adenocarcinoma Tool-Kit (<a href="https://bioconductor.org/packages/release/bioc/html/PDATK.html">PDATK</a>)
<ul>
<li class="fragment">A new Bioconductor package, published by our lab earlier this year</li>
</ul></li>
<li class="fragment">More specifically, we are going to learn how to train and utilize a Pancreatic Cancer Overall Survival Predictor (PCOSP) model</li>
<li class="fragment">The PCOSP model makes predictions to stratify patients into good or bad prognosis
<ul>
<li class="fragment">For this presentation, good means OS &gt;= 365</li>
</ul></li>
<li class="fragment">We hope that this tool may be be useful for other researchers and clinicians</li>
</ul>
<!--
## **What Prerequisite Knowledge Do You Need?** { data-background='images/bhklogo.png' data-background-size='20%' data-background-position='bottom left'}

* A basic understanding of R and R programming
* Familiarity with common Bioconductor classes such as the [`SummarizedExperiment`](https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html)
* Basic understanding of survival analysis and general statistics
* Ability to interpret forest plots and ROC curves
-->
</section>
<section id="presentation-outline" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Presentation Outline</strong></h2>
<ul>
<li class="fragment">Replicate part of a study from our lab, Sandhu <strong>et al.</strong> (2019)
<ol type="1">
<li class="fragment">Learn about how to train a PCOSP model from existing data
<ul>
<li class="fragment">Specifically, RNA sequencing and/or microarray data with known survival outcomes</li>
</ul></li>
<li class="fragment">Use the trained model to make risk predictions for new patient cohorts</li>
<li class="fragment">Demonstrate how to calculate and visualize model performance stats using independent patient cohorts with known survival outcomes</li>
</ol></li>
<li class="fragment">A more complete analysis is available in a <a href="https://bioconductor.org/packages/release/bioc/vignettes/PDATK/inst/doc/PCOSP_model_analysis.html">PDATK vignette</a>, although it only uses a subset of the data</li>
</ul>
</section></section>
<section>
<section id="replicating-sandhu-et-al.-2019" class="title-slide slide level1" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h1 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Replicating Sandhu <em>et al.</em> 2019</strong></h1>

</section>
<section id="the-publication" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>The Publication</strong></h2>
<ul>
<li class="fragment">Meta-Analysis of 1,200 Transcriptomic Profiles Identifies a Prognostic Model for Pancreatic Ductal Adenocarcinoma
<ul>
<li class="fragment">Available at: <a href="https://doi.org/10.1200/cci.18.00102" class="uri">https://doi.org/10.1200/cci.18.00102</a></li>
<li class="fragment">This paper pioneered the PCOSP method</li>
<li class="fragment">Demonstrated it can match or outperform other published signature based risk prediction models</li>
</ul></li>
</ul>
</section>
<section id="the-metagxpancreas-dataset" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>The MetaGxPancreas Dataset</strong></h2>
<ul>
<li class="fragment">Curated and released as the <a href="https://bioconductor.org/packages/release/data/experiment/html/MetaGxPancreas.html">MetaGxPancreas</a> Bioconductor data package
<ul>
<li class="fragment">Contains 15 published pancreatic cancer patient cohorts</li>
<li class="fragment">11 of 15 cohorts have patient survival metadata</li>
</ul></li>
</ul>
</section>
<section id="installload-metagxpancreas-and-pdatk" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Install/Load MetaGxPancreas and PDATK</strong></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>dependencies <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;MetaGxPancreas&#39;</span>, <span class="st">&#39;PDATK&#39;</span>, <span class="st">&#39;BiocParallel&#39;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> dependencies) {</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(pkg)) BiocManager<span class="sc">::</span><span class="fu">install</span>(pkg)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(pkg, <span class="at">character.only=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</section>
<section id="loading-the-datasets" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Loading the Datasets</strong></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: This will download several GB of data to your </span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;ExperimentHub Cache</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>metaGxPanc <span class="ot">&lt;-</span> <span class="fu">loadPancreasDatasets</span>()</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>pancreasCohorts <span class="ot">&lt;-</span> metaGxPanc<span class="sc">$</span>SummarizedExperiments</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>cohortsWithSurvival <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ICGCMICRO&quot;</span>, <span class="st">&quot;ICGCSEQ&quot;</span>, <span class="st">&quot;PCSI&quot;</span>, <span class="st">&quot;TCGA&quot;</span>, </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;KIRBY&quot;</span>, <span class="st">&quot;UNC&quot;</span>, <span class="st">&quot;CHEN&quot;</span>, <span class="st">&quot;COLLISON&quot;</span>, <span class="st">&quot;ZHANG&quot;</span>, <span class="st">&quot;OUH&quot;</span>, <span class="st">&quot;WINTER&quot;</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Fix the names so they match our cohorts</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pancreasCohorts) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&#39;_SumExp&#39;</span>, <span class="fu">names</span>(pancreasCohorts)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to patient cohorts with survival metadata</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>pancreasCohortsWithSurv <span class="ot">&lt;-</span> </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    pancreasCohorts[<span class="fu">grepl</span>(regexCohorts, <span class="fu">names</span>(pancreasCohorts))]</span></code></pre></div>
</section>
<section id="creating-a-cohortlist" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Creating a <code>CohortList</code></strong></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># SurvivalExperiment is just a wrapper around SummarizedExperiment</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;with mandatory survival metadata in colData</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>pancreasSurvExps <span class="ot">&lt;-</span> <span class="fu">lapply</span>(pancreasCohortsWithSurv, </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">FUN=</span>SurvivalExperiment, <span class="at">survival_time=</span><span class="st">&#39;days_to_death&#39;</span>,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">event_occurred=</span><span class="st">&#39;vital_status&#39;</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># A CohortList is just a SimpleList of SurvivalExperiments</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>pancreasCohortList <span class="ot">&lt;-</span> <span class="fu">CohortList</span>(pancreasSurvExps)</span></code></pre></div>
</section>
<section id="split-training-and-validation-data" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Split Training and Validation Data</strong></h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Find common genes</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>commonGenes <span class="ot">&lt;-</span> <span class="fu">findCommonGenes</span>(pancreasCohortList)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># subset on a CohortList subsets all items in the List</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>cohortList <span class="ot">&lt;-</span> <span class="fu">subset</span>(pancreasCohortList, <span class="at">subset=</span>commonGenes)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Train from the ICGC data only</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>ICGCcohorts <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">&#39;ICGC&#39;</span>, <span class="fu">names</span>(cohortList))</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>ICGCcohortList <span class="ot">&lt;-</span> cohortList[ICGCcohorts]</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co"># The remainder of the data will be used for validation</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>validationCohortList <span class="ot">&lt;-</span> cohortList[<span class="sc">!</span>ICGCcohorts]</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Censored patients dropped out of the study before an event occurred</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: This function should be renamed dropCensored</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>validationCohortList <span class="ot">&lt;-</span> <span class="fu">dropNotCensored</span>(validationCohortList)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>ICGCcohortList <span class="ot">&lt;-</span> <span class="fu">dropNotCensored</span>(ICGCcohortList)</span></code></pre></div>
</section>
<section id="split-training-and-validation-data-1" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Split Training and Validation Data</strong></h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Find common samples between the sequencing and array data from ICGC</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>commonSamples <span class="ot">&lt;-</span> <span class="fu">findCommonSamples</span>(ICGCcohortList)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># split into shared samples for training, the rest for testing</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>ICGCtrainCohorts <span class="ot">&lt;-</span> <span class="fu">subset</span>(ICGCcohortList, <span class="at">select=</span>commonSamples)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>ICGCtestCohorts <span class="ot">&lt;-</span> <span class="fu">subset</span>(ICGCcohortList, <span class="at">select=</span>commonSamples, </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">invert=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># merge our training cohort test data into the rest of the validation</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;data</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>validationCohortList <span class="ot">&lt;-</span> <span class="fu">c</span>(ICGCtestCohorts, validationCohortList)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co"># drop ICGCSEQ from the validation data, because it only has 7 </span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;patients, it breaks downstream code</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>validationCohortList <span class="ot">&lt;-</span> </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    validationCohortList[<span class="fu">names</span>(validationCohortList) <span class="sc">!=</span> <span class="st">&#39;ICGCSEQ&#39;</span>]</span></code></pre></div>
</section>
<section id="setup-a-pcosp-model-object" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Setup a <code>PCOSP</code> Model Object</strong></h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set a seed for a reproducible analysis!</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>randomSeed <span class="ot">&lt;-</span> <span class="dv">1987</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(randomSeed)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Construct a PCOSP model object</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>PCOSPmodel <span class="ot">&lt;-</span> <span class="fu">PCOSP</span>(ICGCtrainCohorts, <span class="at">minDaySurvived=</span><span class="dv">365</span>,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">randomSeed=</span>randomSeed)</span></code></pre></div>
<ul>
<li class="fragment">A PCOSP model is a risk prediction method which uses an ensemble of k-top scoring gene pair binary classifiers to vote on a patients prognosis</li>
<li class="fragment">Supervised learning method that uses survival metadata in training data to learn important prognostic features</li>
<li class="fragment">The PCOSP score is defined as the proportion of binary classifiers which vote ‘good’ for prognosis</li>
</ul>
</section>
<section id="training-a-pcosp-model" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Training a <code>PCOSP</code> Model</strong></h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: actually running this requires &gt; 32 GB of RAM if you have </span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;multiple cores try reducing the number of cores in the BiocParallel </span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;defaults or reverting to SerialParam (especially on Windows)</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>trainedPCOSPmodel <span class="ot">&lt;-</span> <span class="fu">trainModel</span>(PCOSPmodel, <span class="at">numModels=</span><span class="dv">1000</span>, </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">minAccuracy=</span><span class="fl">0.6</span>)</span></code></pre></div>
</section>
<section id="risk-prediction-with-a-pcosp-model" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Risk Prediction with a <code>PCOSP</code> Model</strong></h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>PCOSPpredValCohorts <span class="ot">&lt;-</span> <span class="fu">predictClasses</span>(validationCohortList,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">model=</span>trainedPCOSPmodel)</span></code></pre></div>
<ul>
<li class="fragment">This is how you would make prognostic risk predictions with new data
<ul>
<li class="fragment">The first argument to <code>predictClasses</code> should be a <code>SurvivalExperiment</code> or <code>CohortList</code></li>
<li class="fragment">The second argument should be your trained PCOSP model!</li>
</ul></li>
<li class="fragment">If you don’t have your own training data, our pre-trained model can be downloaded from <a href="https://drive.google.com/drive/folders/1E_vztZuUJpck49Vy_zn8v0qVCosFOsNC?usp=sharing">here</a></li>
</ul>
</section>
<section id="risk-prediction-with-a-pcosp-model-1" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Risk Prediction with a <code>PCOSP</code> Model</strong></h2>
<ul>
<li class="fragment">What if we only have rna sequening or microarray data, without survival metadata?
<ul>
<li class="fragment">No problem, we just have to add some fake metadata and risk predictions are still available!</li>
</ul></li>
<li class="fragment">We plan to correct this in a future update, so that a <code>SummarizedExperiment</code> can be used directly</li>
</ul>
</section>
<section id="risk-prediction-with-a-pcosp-model-2" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Risk Prediction with a <code>PCOSP</code> Model</strong></h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>sumExp <span class="ot">&lt;-</span> <span class="fu">as</span>(validationCohortList[[<span class="st">&#39;UNC&#39;</span>]], <span class="st">&#39;SummarizedExperiment&#39;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>sumExp</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 12424 109 
## metadata(6): experimentData annotation ... PCOSPpredictions PCOSPparams
## assays(1): exprs
## rownames(12424): A1CF A2M ... ZZEF1 ZZZ3
## rowData names(4): gene EntrezGene.ID Ensemble.ID best_probe
## colnames(109): X46337 X46339 ... X72613 X72616
## colData names(7): unique_patient_ID survival_time ... prognosis
##   PCOSP_prob_good</code></pre>
<ol type="1">
<li class="fragment">Load your data as a <code>SummarizedExperiment</code></li>
</ol>
</section>
<section id="risk-prediction-with-a-pcosp-model-3" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Risk Prediction with a <code>PCOSP</code> Model</strong></h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(sumExp)<span class="sc">$</span>survival_time <span class="ot">&lt;-</span> 0L</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(sumExp)<span class="sc">$</span>event_occurred <span class="ot">&lt;-</span> 0L</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>survExp <span class="ot">&lt;-</span> <span class="fu">SurvivalExperiment</span>(sumExp)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>survExp</span></code></pre></div>
<pre><code>## class: SurvivalExperiment 
## dim: 12424 109 
## metadata(6): experimentData annotation ... PCOSPpredictions PCOSPparams
## assays(1): exprs
## rownames(12424): A1CF A2M ... ZZEF1 ZZZ3
## rowData names(4): gene EntrezGene.ID Ensemble.ID best_probe
## colnames(109): X46337 X46339 ... X72613 X72616
## colData names(7): unique_patient_ID survival_time ... prognosis
##   PCOSP_prob_good</code></pre>
<ol start="2" type="1">
<li class="fragment">Add some fake metadata and convert to a <code>SurvivalExperiment</code></li>
</ol>
</section>
<section id="risk-prediction-with-a-pcosp-model-4" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Risk Prediction with a <code>PCOSP</code> Model</strong></h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>riskPredictions <span class="ot">&lt;-</span> <span class="fu">predictClasses</span>(survExp, <span class="at">model=</span>trainedPCOSPmodel)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(riskPredictions)[, <span class="fu">c</span>(<span class="st">&#39;sample_name&#39;</span>, <span class="st">&#39;PCOSP_prob_good&#39;</span>)]</span></code></pre></div>
<ol start="3" type="1">
<li class="fragment">Make your risk predictions!</li>
</ol>
<ul>
<li class="fragment">For our validation calculations, we assume a PCOSP score &gt; 0.5 predicts good prognosis
<ul>
<li class="fragment">This is required to ensure we maximize the statistical power of our tests</li>
</ul></li>
<li class="fragment">In clinical applicatons, a stricter cut-off may provide more certainty of prognosis
<ul>
<li class="fragment">Could be a good follow up study!</li>
</ul></li>
</ul>
</section>
<section id="risk-prediction-with-a-pcosp-model-5" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Risk Prediction with a <code>PCOSP</code> Model</strong></h2>
<div id="htmlwidget-d460a30919ea1fad2aa6" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d460a30919ea1fad2aa6">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"sample_name":["X46337","X46339","X46450","X46452","X46460","X46578","X46581","X46582","X46584","X46585","X46587","X46592","X46642","X46643","X46644","X46645","X46646","X46647","X46648","X46649","X46650","X46651","X46652","X46653","X46826","X46828","X46830","X46831","X46832","X46985","X46986","X47590","X47692","X47695","X47700","X47701","X47702","X47703","X47708","X47965","X47969","X47983","X47989","X48550","X48556","X48558","X48562","X48564","X48567","X48569","X49388","X49392","X52042","X52043","X54171","X54172","X54173","X54174","X54175","X54243","X54291","X54292","X54293","X54294","X54295","X54296","X54297","X54299","X54300","X54301","X54302","X54305","X54306","X54307","X54308","X54309","X54310","X54311","X54312","X54313","X54314","X54315","X54318","X56367","X56525","X56527","X56528","X56530","X56536","X56537","X56538","X56539","X56540","X64482","X64492","X64494","X64496","X64497","X64498","X64500","X64502","X64503","X64504","X64505","X64508","X64509","X64510","X72613","X72616"],"PCOSP_prob_good":[0.540590405904059,0.269372693726937,0.522140221402214,0.47970479704797,0.300738007380074,0.712177121771218,0.6309963099631,0.525830258302583,0.629151291512915,0.507380073800738,0.464944649446494,0.413284132841328,0.485239852398524,0.330258302583026,0.488929889298893,0.498154981549816,0.527675276752767,0.452029520295203,0.654981549815498,0.710332103321033,0.621771217712177,0.551660516605166,0.494464944649447,0.284132841328413,0.498154981549816,0.485239852398524,0.61070110701107,0.175276752767528,0.511070110701107,0.431734317343173,0.682656826568266,0.437269372693727,0.300738007380074,0.483394833948339,0.730627306273063,0.450184501845018,0.612546125461255,0.354243542435424,0.619926199261993,0.728782287822878,0.621771217712177,0.529520295202952,0.453874538745387,0.468634686346863,0.188191881918819,0.481549815498155,0.588560885608856,0.533210332103321,0.498154981549816,0.42619926199262,0.632841328413284,0.57380073800738,0.690036900369004,0.53690036900369,0.645756457564576,0.747232472324723,0.40590405904059,0.476014760147601,0.57380073800738,0.477859778597786,0.531365313653137,0.612546125461255,0.59409594095941,0.540590405904059,0.40590405904059,0.584870848708487,0.496309963099631,0.551660516605166,0.542435424354244,0.533210332103321,0.40590405904059,0.599630996309963,0.411439114391144,0.564575645756458,0.46309963099631,0.583025830258303,0.616236162361624,0.549815498154982,0.520295202952029,0.470479704797048,0.477859778597786,0.24169741697417,0.505535055350554,0.640221402214022,0.435424354243542,0.402214022140221,0.518450184501845,0.61070110701107,0.243542435424354,0.533210332103321,0.603321033210332,0.634686346863469,0.57380073800738,0.555350553505535,0.501845018450184,0.42619926199262,0.774907749077491,0.525830258302583,0.313653136531365,0.529520295202952,0.433579335793358,0.359778597785978,0.678966789667897,0.53690036900369,0.514760147601476,0.324723247232472,0.466789667896679,0.664206642066421,0.514760147601476]},"columns":[{"accessor":"sample_name","name":"Sample","type":"character"},{"accessor":"PCOSP_prob_good","name":"PCOSP Score","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"defaultPageSize":10,"paginationType":"simple","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"dataKey":"85ef8a0231598a7b9f380b4184175b2c","key":"85ef8a0231598a7b9f380b4184175b2c"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</section>
<section id="risk-prediction-with-a-pcosp-model-6" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Risk Prediction with a <code>PCOSP</code> Model</strong></h2>
<ul>
<li class="fragment">To recap, predicting risk for new patient cohorts:
<ol type="1">
<li class="fragment">Load your data as a <code>SummarizedExperiment</code></li>
<li class="fragment">Add some fake metadata and convert to a <code>SurvivalExperiment</code></li>
<li class="fragment">Make your risk predictions!</li>
</ol></li>
<li class="fragment">What to do with the results is the interesting question!</li>
<li class="fragment">Potential for more research:
<ul>
<li class="fragment">Applications to other cancer types (PCOSP -&gt; COSP)</li>
<li class="fragment">Development of target sequencing panel for the PCOSP gene signature</li>
<li class="fragment">Clinical validation of PCOSP score predictions in prospective cohort</li>
</ul></li>
</ul>
</section>
<section id="validating-a-pcosp-model" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Validating a <code>PCOSP</code> Model</strong></h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>validatedPCOSPmodel <span class="ot">&lt;-</span> <span class="fu">validateModel</span>(trainedPCOSPmodel,</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">valData=</span>PCOSPpredValCohorts)</span></code></pre></div>
<ul>
<li class="fragment">This performs external validation using the remaining ICGCMICRO samples, as well as the nine remaining independent patient cohorts</li>
<li class="fragment">A <code>data.table</code> of performance statistics can be accessed via the <code>validationStats</code> PCOSP-method</li>
<li class="fragment">This is the table which is used to generate all subsequent plots</li>
</ul>
</section>
<section id="plotting-pcosp-model-performance" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Plotting <code>PCOSP</code> Model Performance</strong></h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>PCOSPdIndexForestPlot <span class="ot">&lt;-</span> <span class="fu">forestPlot</span>(validatedPCOSPmodel, </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">stat=</span><span class="st">&#39;log_D_index&#39;</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>PCOSPdIndexForestPlot</span></code></pre></div>
<ul>
<li class="fragment">D-index is the log2 Hazard ratio, a measure of relative risk
<ul>
<li class="fragment">A higher D-index is better, as it represents a larger difference between the good and bad prognosis groups</li>
<li class="fragment">0 is the cut-off for better prognosis in the good survival group vs the bad</li>
</ul></li>
</ul>
</section>
<section id="plotting-pcosp-model-performance-1" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Plotting <code>PCOSP</code> Model Performance</strong></h2>
<p><img src="PancreaticCancerOverallSurvivalPredictor_files/figure-revealjs/actual_pcosp_forest_plot1-1.png" width="768" /></p>
</section>
<section id="plotting-pcosp-model-performance-2" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Plotting <code>PCOSP</code> Model Performance</strong></h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>PCOSPconcIndexForestPlot <span class="ot">&lt;-</span> <span class="fu">forestPlot</span>(validatedPCOSPmodel, </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">stat=</span><span class="st">&#39;concordance_index&#39;</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>PCOSPconcIndexForestPlot</span></code></pre></div>
<ul>
<li class="fragment">Concordance index is a measure of similarity between rankings
<ul>
<li class="fragment">In this case, the ranking is based on prognostic risk</li>
<li class="fragment">The ranking achieved by the PCOSP score is compared to real observed survival data</li>
<li class="fragment">A concordance index greater than 0.5 indicates better than random performance</li>
<li class="fragment">A concordance index of 1 indicates identical ranking</li>
</ul></li>
</ul>
</section>
<section id="plotting-pcosp-model-performance-3" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Plotting <code>PCOSP</code> Model Performance</strong></h2>
<p><img src="PancreaticCancerOverallSurvivalPredictor_files/figure-revealjs/actual_pcosp_forest_plot2-1.png" width="768" /></p>
</section>
<section id="plotting-pcosp-model-performance-4" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Plotting <code>PCOSP</code> Model Performance</strong></h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>cohortROCplots <span class="ot">&lt;-</span> <span class="fu">plotROC</span>(validatedPCOSPmodel, <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>cohortROCplots</span></code></pre></div>
<ul>
<li class="fragment">ROC curves show the performance of a classification model</li>
<li class="fragment">An ideal model would have its curve as far to the top left as possible</li>
<li class="fragment">The <code>y = x</code> line represents statistical insignificance, that the model did no better than random chance</li>
</ul>
</section>
<section id="plotting-pcosp-model-performance-5" class="slide level2" data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left">
<h2 data-background="images/bhklogo.png" data-background-size="20%" data-background-position="bottom left"><strong>Plotting <code>PCOSP</code> Model Performance</strong></h2>
<p><img src="PancreaticCancerOverallSurvivalPredictor_files/figure-revealjs/actual_pcosp_roc_curves-1.png" width="768" /></p>
</section></section>
<section>
<section id="questions" class="title-slide slide level1">
<h1>Questions?</h1>

</section>
<section id="references" class="slide level2">
<h2>References</h2>
<ol type="1">
<li class="fragment">Michael Zon, Vandana Sandhu and Benjamin Haibe-Kains (2021). MetaGxPancreas: Transcriptomic Pancreatic Cancer Datasets. R package version 1.13.0.</li>
<li class="fragment">Sandhu V, Labori KJ, Borgida A, Lungu I, Bartlett J, Hafezi-Bakhtiari S, Denroche RE, Jang GH, Pasternack D, Mbaabali F, Watson M, Wilson J, Kure EH, Gallinger S, Haibe-Kains B. Meta-Analysis of 1,200 Transcriptomic Profiles Identifies a Prognostic Model for Pancreatic Ductal Adenocarcinoma. JCO Clin Cancer Inform. 2019 May;3:1-16. doi: 10.1200/CCI.18.00102. PMID: 31070984.</li>
</ol>
</section></section>
    </div>
  </div>

  <script src="PancreaticCancerOverallSurvivalPredictor_files/reveal.js-3.3.0.1/lib/js/head.min.js"></script>
  <script src="PancreaticCancerOverallSurvivalPredictor_files/reveal.js-3.3.0.1/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
        // Push each slide change to the browser history
        history: true,
        // Vertical centering of slides
        center: true,
        // Opens links in an iframe preview overlay
        previewLinks: true,
        // Transition style
        transition: 'default', // none/fade/slide/convex/concave/zoom
        // Transition style for full page slide backgrounds
        backgroundTransition: 'default', // none/fade/slide/convex/concave/zoom



        // Optional reveal.js plugins
        dependencies: [
        ]
      });
    </script>
  <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
      var script = document.createElement("script");
      script.type = "text/javascript";
      script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
      document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>

<script>
  (function() {
    if (window.jQuery) {
      Reveal.addEventListener( 'slidechanged', function(event) {
        window.jQuery(event.previousSlide).trigger('hidden');
        window.jQuery(event.currentSlide).trigger('shown');
      });
    }
  })();
</script>


  </body>
</html>
